extensions/positron-python/python_files/positron/positron_ipykernel/ui.py

```diff
@@ -216,12 +208,48 @@ def open(self, url, new=0, autoraise=True):
                     height=0,
                 ).dict(),
             )
-            return True
 
         for addr in _localhosts:
             if addr in url:
-                event = ShowUrlParams(url=url)
-                self._comm.send_event(name=UiFrontendEvent.ShowUrl, payload=event.dict())
+                is_plot = self._is_module_function("plotly.basedatatypes", "show")
+                if is_plot:
+                    if os.name == "nt":
+                        url = urlparse(url).netloc or urlparse(url).path
+                    event = ShowUrlParams(url=url)
+                    self._send_show_html_event(url, is_plot)
+                else:
+                    event = ShowUrlParams(url=url)
+                    self._comm.send_event(name=UiFrontendEvent.ShowUrl, payload=event.dict())
+
                 return True
         # pass back to webbrowser's list of browsers to open up the link
         return False
+
+
+    def _is_module_function(self, module_name, function_name):
+        module = sys.modules.get(module_name)
+        if module:
+            for frame_info in inspect.stack():
+                if (
+                    inspect.getmodule(frame_info.frame, frame_info.filename) == module
```

Comment:
I initially wrote this code, and I've wondered if it would be safer to check only the module part and not the function. What do you think?

Reply:
It seemed to make sense to check the function too but I don't think I know enough about Python to foresee any problems. I suppose removing the function check would handle any new calls that Bokeh or Plotly might add.

Then it could work in that scenario without having to do anything. Keeping the check would definitely not work in that case so there's nothing to lose?